sdk/core/core/test/cmocka/test_span.c

```diff
@@ -81,138 +84,131 @@ static void az_single_char_ascii_lower_test()
-  assert_int_equal(az_span_length(buffer), 1);
-  buffer = az_span_append_uint8(buffer, 'b');
-  assert_int_equal(az_span_length(buffer), 2);
-  buffer = az_span_append_uint8(buffer, 'c');
-  assert_int_equal(az_span_length(buffer), 3);
+  buffer = az_span_copy_uint8(buffer, 'a');
+  assert_int_equal(az_span_size(buffer), 14);
+  buffer = az_span_copy_uint8(buffer, 'b');
+  assert_int_equal(az_span_size(buffer), 13);
+  buffer = az_span_copy_uint8(buffer, 'c');
+  assert_int_equal(az_span_size(buffer), 12);
 
-  assert_true(az_span_is_content_equal(buffer, AZ_SPAN_FROM_STR("abc")));
+  assert_true(az_span_is_content_equal(
+      az_span_slice(AZ_SPAN_FROM_BUFFER(raw_buffer), 0, 3), AZ_SPAN_FROM_STR("abc")));
 }
 
-static void az_span_append_i32toa_succeeds()
+static void az_span_itoa_succeeds()
 {
   int32_t v = 12345;
   uint8_t raw_buffer[15];
   az_span buffer = AZ_SPAN_FROM_BUFFER(raw_buffer);
   az_span out_span;
 
-  assert_true(az_succeeded(az_span_append_i32toa(buffer, v, &out_span)));
-  assert_true(az_span_is_content_equal(out_span, AZ_SPAN_FROM_STR("12345")));
+  assert_true(az_succeeded(az_span_itoa(buffer, v, &out_span)));
+  assert_int_equal(az_span_size(out_span), 10);
+  assert_true(az_span_is_content_equal(
+      az_span_slice(AZ_SPAN_FROM_BUFFER(raw_buffer), 0, 5), AZ_SPAN_FROM_STR("12345")));
 }
 
-static void az_span_append_i32toa_negative_succeeds()
+static void az_span_itoa_negative_succeeds()
 {
   int32_t v = -12345;
   uint8_t raw_buffer[15];
   az_span buffer = AZ_SPAN_FROM_BUFFER(raw_buffer);
   az_span out_span;
 
-  assert_true(az_succeeded(az_span_append_i32toa(buffer, v, &out_span)));
-  assert_true(az_span_is_content_equal(out_span, AZ_SPAN_FROM_STR("-12345")));
+  assert_true(az_succeeded(az_span_itoa(buffer, v, &out_span)));
+  assert_int_equal(az_span_size(out_span), 9);
+  assert_true(az_span_is_content_equal(
+      az_span_slice(AZ_SPAN_FROM_BUFFER(raw_buffer), 0, 6), AZ_SPAN_FROM_STR("-12345")));
 }
 
-static void az_span_append_i32toa_zero_succeeds()
+static void az_span_itoa_zero_succeeds()
 {
   int32_t v = 0;
   uint8_t raw_buffer[15];
   az_span buffer = AZ_SPAN_FROM_BUFFER(raw_buffer);
   az_span out_span;
 
-  assert_true(az_succeeded(az_span_append_i32toa(buffer, v, &out_span)));
-  assert_true(az_span_is_content_equal(out_span, AZ_SPAN_FROM_STR("0")));
-}
-
-static void az_span_append_i32toa_max_int_succeeds()
-{
-  int32_t v = 2147483647;
-  uint8_t raw_buffer[15];
-  az_span buffer = AZ_SPAN_FROM_BUFFER(raw_buffer);
-  az_span out_span;
-
-  assert_true(az_succeeded(az_span_append_i32toa(buffer, v, &out_span)));
-  assert_true(az_span_is_content_equal(out_span, AZ_SPAN_FROM_STR("2147483647")));
+  assert_true(az_succeeded(az_span_itoa(buffer, v, &out_span)));
+  assert_int_equal(az_span_size(out_span), 14);
+  assert_true(az_span_is_content_equal(
+      az_span_slice(AZ_SPAN_FROM_BUFFER(raw_buffer), 0, 1), AZ_SPAN_FROM_STR("0")));
 }
 
-static void az_span_append_i32toa_NULL_span_fails()
```

Comment:
This was a duplicate of `az_span_append_i32toa_max_int_succeeds` and wasn't testing `NULL`. Changing this test would hit the precondition, so we should add it back in the future with other precondition tests.